src/V3EmitCImp.cpp

```diff
@@ -888,19 +901,49 @@ void V3EmitC::emitcImp() {
     UINFO(2, __FUNCTION__ << ": " << endl);
     // Make parent module pointers available.
     const EmitCParentModule emitCParentModule;
+    std::list<std::future<void>> m_futures;
+    std::deque<std::shared_ptr<std::deque<AstCFile*>>> m_filenames;
```

Comment:
Just use list as the outer container, emplace_back the new dequeue and pass the back of the list be reference to the visitors. I don't think there is real need for the shared_ptr here.